src/CoreBundle/private/library/classes/TCMSResourceCollection.class.php

```diff
@@ -536,6 +536,23 @@ public function ReplaceRealPathCallback($aMatch)
         return $aMatch[0];
     }
 
+    private function isRelativeUrl(string $url): bool
+    {
+        if ('/' === substr($url, 0, 1)) {
+            return false;
+        }
+
+        if ('http://' === substr($url, 0, 7) || 'https://' === substr($url, 0, 8)) {
+            return false;
+        }
+
+        if ('data:' === substr($url, 0, 5)) {
```

Comment:
Shouldn't this be filtered earlier?
ie in \TCMSResourceCollection::ReplaceRelativePath() in the regex?

Because "data:" is no url to start with.

Reply:
Changing the regex may have side effects. I'm not a regex guru. But yes, in theory you could handle all of this in the regex already. If you know how exactly and be sure this doesn't have side effects.